### PR TITLE
Add Guide for Flutter FCM Novu Push Notification

### PR DESCRIPTION
**Pull request by unicodeveloper**

### CODE REVIEW

File: guides/fcm-flutter-novu/how-to-send-push-notifications-to-flutter-apps-with-fcm-using-Novu.mdx [hunk 1]
```diff
@@ -0,0 +1,735 @@
...
+- [Android Studio](https://developer.android.com/studio/install) configured with Dart and Flutter plugins.
+- [Xcode](https://developer.apple.com/xcode/) installed on your machine.
+
+Check out the [Flutter app code for this guide on GitHub](https://github.com/novuhq/flutter-fcm-novu).
+
+## Set up and Create a Flutter App
+
+If you have an existing Flutter App, you can skip this step.
+
+If you are creating a brand new Flutter app for the first time, please follow this [guide](https://firebase.google.com/codelabs/firebase-fcm-flutter#0).
```
Comment (sumitsaurabh927) [inline guides/fcm-flutter-novu/how-to-send-push-notifications-to-flutter-apps-with-fcm-using-Novu.mdx:24]: I feel that "for the first time" can be removed here as it makes it sound as if this is the very first flutter app one is creating

File: guides/fcm-flutter-novu/how-to-send-push-notifications-to-flutter-apps-with-fcm-using-Novu.mdx [hunk 2]
```diff
@@ -0,0 +1,735 @@
...
+
+## Set up Firebase and FlutterFire
+
+Please follow this [guide to set up a Firebase project and integrate Firebase Cloud Messaging in your Flutter app](https://firebase.google.com/codelabs/firebase-fcm-flutter#2).
+
+## Set Up Apple Integration
+
+Please follow this [guide to ensure everything is set up on your iOS devices](https://firebase.flutter.dev/docs/messaging/apple-integration) to receive messages.
```
Comment (sumitsaurabh927) [inline guides/fcm-flutter-novu/how-to-send-push-notifications-to-flutter-apps-with-fcm-using-Novu.mdx:32]: 1. guide to ensure => guide to ensure that
2. devices => device

File: guides/fcm-flutter-novu/how-to-send-push-notifications-to-flutter-apps-with-fcm-using-Novu.mdx [hunk 3]
```diff
@@ -0,0 +1,735 @@
...
+  }
+}
+```
+_lib/main.dart_
+
+There are 4 sections to pay attention to:
+
+<Steps>
+  <Step title="The Request permission code block">
+    This shows the notification permission prompt to the user the first time the user interacts with the app. If the user allows it, then we can get a token.
```
Comment (sumitsaurabh927) [inline guides/fcm-flutter-novu/how-to-send-push-notifications-to-flutter-apps-with-fcm-using-Novu.mdx:186]: This shows the notification permission prompt to the user the first time the user interacts with the app => This shows the notification permission prompt to the user when the user interacts with the app for the first time

File: guides/fcm-flutter-novu/how-to-send-push-notifications-to-flutter-apps-with-fcm-using-Novu.mdx [hunk 4]
```diff
@@ -0,0 +1,735 @@
...
+
+## Set Up Apple Integration
+
+Please follow this [guide to ensure everything is set up on your iOS devices](https://firebase.flutter.dev/docs/messaging/apple-integration) to receive messages.
+
+There is an optional section about allowing **Notification Images**. Follow the steps to install and activate the notification service extension. We'll need it later in this guide to display images as part of the push notifications from FCM.
+
+## Set Up FCM Notification Permission & Registration Token
+
+We'll add code to initialize firebase and ask the user's permission for receiving notifications.
```
Comment (sumitsaurabh927) [inline guides/fcm-flutter-novu/how-to-send-push-notifications-to-flutter-apps-with-fcm-using-Novu.mdx:38]: 1. firebase => Firebase
2. for receiving notifications => to receive notifications

File: guides/fcm-flutter-novu/how-to-send-push-notifications-to-flutter-apps-with-fcm-using-Novu.mdx [hunk 5]
```diff
@@ -0,0 +1,735 @@
...
+    ```dart
+      String? token = await messaging.getToken();
+
+      if (kDebugMode) {
+        print('Registration Token=$token');
+      }
+    ```
+  </Step>
+  <Step title="The foreground message handler">
+    This handler listens to when a push notification is fired from FCM to the device. If the app is in foreground, then it prints out the notification title, body, messageId and data properties to the console.
```
Comment (sumitsaurabh927) [inline guides/fcm-flutter-novu/how-to-send-push-notifications-to-flutter-apps-with-fcm-using-Novu.mdx:217]: 1. This handler listens to when a push notification is fired from FCM to the device => This handler is triggered when a push notification is sent from FCM to the device.
2. If app is in foreground => if app is in the foreground
3. messageId and data => messageId, and data

File: guides/fcm-flutter-novu/how-to-send-push-notifications-to-flutter-apps-with-fcm-using-Novu.mdx [hunk 6]
```diff
@@ -0,0 +1,735 @@
...
+          print('Message notification: ${message.notification?.title}');
+          print('Message notification: ${message.notification?.body}');
+        }
+
+        _messageStreamController.sink.add(message);
+      });
+    ```
+  </Step>
+  <Step title="The background message handler">
+    This handler listens to when a push notification is fired from FCM to the device. If the app is in background, then it prints out the notification title, body, messageId and data properties to the console.
```
Comment (sumitsaurabh927) [inline guides/fcm-flutter-novu/how-to-send-push-notifications-to-flutter-apps-with-fcm-using-Novu.mdx:233]: 1. This handler listens to when a push notification is fired from FCM to the device => This handler is triggered when a push notification is fired from FCM to the device.
2. If app is in background => if app is in the background
3. messageId and data => messageId, and data

File: guides/fcm-flutter-novu/how-to-send-push-notifications-to-flutter-apps-with-fcm-using-Novu.mdx [hunk 7]
```diff
@@ -0,0 +1,735 @@
...
+      }
+      ...
+
+
+      FirebaseMessaging.onBackgroundMessage(_firebaseMessagingBackgroundHandler);
+    ```
+  </Step>
+</Steps>
+
+Now, run the Flutter app. Your device should be connected to your machine to enable the app run on it.
```
Comment (sumitsaurabh927) [inline guides/fcm-flutter-novu/how-to-send-push-notifications-to-flutter-apps-with-fcm-using-Novu.mdx:257]: the app run on it => the app to run on it

File: guides/fcm-flutter-novu/how-to-send-push-notifications-to-flutter-apps-with-fcm-using-Novu.mdx [hunk 8]
```diff
@@ -0,0 +1,735 @@
...
+```
+
+**Trigger a workflow via the Novu Dashboard**
+
+<Frame caption="Add Push title and content"><img src="https://github.com/novuhq/docs/assets/2946769/e6f5aefb-4abe-4190-8d94-16f31650427d"/> </Frame>
+<Frame caption="Trigger Workflow"><img src="https://github.com/novuhq/docs/assets/2946769/cfa2e904-f07a-4796-9020-6917fdb17aa8"/> </Frame>
+
+**Receive Triggered Push Workflow on Device**
+
+<Frame caption="Push Notification Received on Device">
```
Comment (sumitsaurabh927) [inline guides/fcm-flutter-novu/how-to-send-push-notifications-to-flutter-apps-with-fcm-using-Novu.mdx:472]: on device => on the device

File: guides/fcm-flutter-novu/how-to-send-push-notifications-to-flutter-apps-with-fcm-using-Novu.mdx [hunk 9]
```diff
@@ -0,0 +1,735 @@
...
+              "subscription_status": "active",
+              "renewed_by": "bisola"
+            }
+          }
+         }
+       }'
+
+```
+
+<Frame caption="Push & Data Notification received on device in foreground from Novu via FCM">
```
Comment (sumitsaurabh927) [inline guides/fcm-flutter-novu/how-to-send-push-notifications-to-flutter-apps-with-fcm-using-Novu.mdx:547]: Notification received on device in foreground => Notification received on device in the foreground

**Review by sumitsaurabh927 — CHANGES_REQUESTED** [review]
Hey, @unicodeveloper dropped some comments. 
Great work as always! 💪

File: guides/fcm-flutter-novu/how-to-send-push-notifications-to-flutter-apps-with-fcm-using-Novu.mdx (hunk 1 above)
Comment (unicodeveloper) [reply guides/fcm-flutter-novu/how-to-send-push-notifications-to-flutter-apps-with-fcm-using-Novu.mdx:24]: Yeah, the reason I added it is if the reader has never created a new flutter app ever before. But I will remove it.

**Pull request merged**